components/StorageFileListItem.vue

```diff
@@ -14,13 +14,22 @@ const emit = defineEmits<{
   (e: "updateFileList"): void;
 }>();
 
+const { storage } = useStorage();
+
 const layoutType = useLayoutType();
-const getUrl = useRetrievePublicUrl(props.file.name);
+const getPublicUrl = useRetrievePublicFileUrl(props.file.name);
+// const getPrivateUrl = useRetrievePrivateFileUrl(props.file.name);
 const selectedFiles = useSelectedFiles();
 const sortType = useSortType(props.file);
 
 const previewUrl = computed(() => {
-  return getUrl.url.value?.publicUrl;
+  if (storage.value.bucket === "private") {
+    const getPrivateUrl = useRetrievePrivateFileUrl(props.file.name);
+    return getPrivateUrl.privateUrl.value;
+  } else {
+    // const getPublicUrl = useRetrievePublicFileUrl(props.file.name);
+    return getPublicUrl.url.value?.publicUrl;
+  }
 });
```

Comment:
Here I have a problem with getting the url of file to get file preview in the previewUrl composable. So it works this way, but anyway I get vue warnings in the console. Tried to use the same approach for private and public storage - but would not get the file preview. Maybe you have some ideas.

Reply:
I played with that - and what I found is that `getPublicUrl` function inside `useRetrievePublicFileUrl` should be a sync function and there is no need to use `useAsyncData` composable here 👌 here all my changes:
<img width="578" alt="Zrzut ekranu 2023-10-5 o 16 34 51" src="https://github.com/MartaSolo/nuxt3-playground/assets/72549525/cb1e6b2e-2ec0-4003-9472-0de3ae50727f">
<img width="636" alt="Zrzut ekranu 2023-10-5 o 16 35 19" src="https://github.com/MartaSolo/nuxt3-playground/assets/72549525/1f7abccc-b7e8-44fd-b88b-e1aaaf082195">

Reply:
you can ofc return just string from composable instead of `{ url }` object and it might be even simpler 👌

Reply:
Thank you :), in the composable when I try to  `return { data.publicUrl }`; I get the ts error: ` ',' expected.ts(1005)`, so I will leave it.

The other composable to get private files -  useRetrievePrivateFileUrl.ts is async function so I didn't edit it to be as the public one.